examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink (stv 1 0.99999982)
+            (PredicateNode "contain")
+            (ListLink
+               (ConceptNode "treatment-1")
+               (ConceptNode "compound-A")
+            )
+         )
+      )
+   )
+...
+scheme@(guile-user)> (cog-bind lambda-grounded-construction-rule)
+$5 = (SetLink
+   (LambdaLink (stv 1 0.99999982)
+      (TypedVariableLink
+         (VariableNode "$X")
+         (TypeNode "ConceptNode")
+      )
+      (EvaluationLink (stv 1 0.99999982)
+         (PredicateNode "contain")
+         (ListLink
+            (ConceptNode "treatment-1")
+            (ConceptNode "compound-A")
+         )
+      )
    )
 )
 
+scheme@(guile-user)> (cog-bind implication-construction-rule)
+...
+   (ImplicationLink (stv 1 0.99999982)
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink
+            (PredicateNode "take")
+            (ListLink
+               (VariableNode "$X")
+               (ConceptNode "treatment-1")
+            )
+         )
+      )
+      (LambdaLink (stv 1 0.99999982)
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink (stv 1 0.99999982)
+            (PredicateNode "contain")
+            (ListLink
+               (ConceptNode "treatment-1")
+               (ConceptNode "compound-A")
+            )
+         )
+      )
+   )
+...
+
+scheme@(guile-user)> (cog-bind implication-implicant-distribution-rule)
+...
+   (ImplicationLink (stv 1 0.99999982)
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink
+            (PredicateNode "take")
+            (ListLink
+               (VariableNode "$X")
+               (ConceptNode "treatment-1")
+            )
+         )
+      )
+      (AndLink
+         (LambdaLink
+            (TypedVariableLink
+               (VariableNode "$X")
+               (TypeNode "ConceptNode")
+            )
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+         )
+         (LambdaLink (stv 1 0.99999982)
+            (TypedVariableLink
+               (VariableNode "$X")
+               (TypeNode "ConceptNode")
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+   )
+...
+scheme@(guile-user)> ;; If X takes treatment-1 then X takes compound-A
+scheme@(guile-user)> (cog-bind deduction-implication-rule)
```

Comment:
well, again, this seems very awkward, because X is free in the english-language phrase, whereas in the below, there are two different X's that are both lambda-bound below. --  there are two distinct x's which can be alpha-converted to any letter, say, y and z.   So, below, what is being said is that  "double-turnstile (its a semantic consequence that) [(lambda.y) p(y) -> lambda.z q(z) ]"  where p(y) is  "y takes treatment-1" and q(z) is "z takes compound-A".  By using the same symbol x in both lambdas, its confusing and misleading.

By contrast, in the statement above, "If X takes treatment-1 then X takes compound-A" the variable x is free, and not bound!   The only way to  make it free is to to say  "double-turnstile (its a semantic consequence that) [(lambda.y) p(y) -> lambda.z q(z) ] (x)"  which can then be beta-reduced to "  "double-turnstile (its a semantic consequence that) [p(x) ->q(x)]" so that x is now free.

Alternately you can turn this back into a combinator by binding x again: "double-turnstile (its a semantic consequence that) lambda.x [p(x) ->q(x)]"   

To summarize, the english-language explanation is a predicate-logic statement with free variables, but the actual code is some combinators with no free variables.  This is confusing because there's a lot of computation and juggling you have to do in your head to convert the one to the other.   Better to write it out ...